services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -1342,4 +1394,636 @@ public ResponseEntity<Page<MinuteCount>> getBsmActivityByMinuteInRange(
         return ResponseEntity
                 .ok(new PageImpl<>(outputEvents, PageRequest.of(page, size), pagedEvents.getTotalElements()));
     }
+
+    @Operation(summary = "Find Spat Minimum Data Event Aggregations", description = "Returns a list of Spat Minimum Data Event Aggregations, filtered by intersection ID, start time, end time, and latest. The latest parameter will return the most recent message satisfying the query.")
+    @RequestMapping(value = "/events/spat_minimum_data_event_aggregation", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID, 'USER') and @PermissionService.hasRole('USER')) ")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Success"),
+            @ApiResponse(responseCode = "403", description = "Forbidden - Requires SUPER_USER or USER role with access to the intersection requested")
+    })
+    public ResponseEntity<Page<SpatMinimumDataEventAggregation>> findSpatMinimumDataEventAggregation(
+            @RequestParam(name = "intersection_id") Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "page", required = false, defaultValue = "0") int page,
+            @RequestParam(name = "size", required = false, defaultValue = "10000") int size,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+        if (testData) {
+            // Mock response for test data
+            List<SpatMinimumDataEventAggregation> mockList = List
+                    .of(MockAggregatedEventGenerator.getSpatMinimumDataEventAggregation());
+            Page<SpatMinimumDataEventAggregation> mockPage = new PageImpl<>(mockList, PageRequest.of(page, size),
+                    mockList.size());
+            return ResponseEntity.ok(mockPage);
+        } else if (latest) {
+            return ResponseEntity.ok(spatMinimumDataEventAggregationRepo.findLatest(intersectionID,
+                    startTime, endTime));
+        } else {
+            // Retrieve a paginated result from the repository
+            PageRequest pageable = PageRequest.of(page, size);
+            Page<SpatMinimumDataEventAggregation> response = spatMinimumDataEventAggregationRepo.find(intersectionID,
+                    startTime, endTime, pageable);
+            log.debug("Returning SpatMinimumDataEventAggregation Page with Size: {}", response.getContent().size());
```

Comment:
I have been removing these, I have a slight preference for removing them, but I don't mind keeping them if they have a purpose